### PR TITLE
Have the benchmark role store metadata about run

### PR DESCRIPTION
**Pull request by jtaleric**
We need to have a mechanism to determine *what* the benchmark SUT was at
the time of the run. Instead of having to look back at metadata, we
should have a single index which describes the SUT.

Signed-off-by: Joe Talerico <rook@isovalent.com>

### CODE REVIEW

File: roles/benchmarks/tasks/run-benchmark.yml [hunk 1]
```diff
@@ -1,67 +1,90 @@
...
+    var: report.stdout
+
+- name: Cleanup Run
+  shell: |
+    export KUBECONFIG={{kubeconfig}}
+    kubectl delete benchmark/{{item.name}} -n benchmark-operator
+  when: benchmark_cleanup
+
+- name: Build metadata payload
+  template:
```
Comment (learnitall) [inline roles/benchmarks/tasks/run-benchmark.yml:74]: This is a super great idea, using the template to create the payload. Love it.

File: roles/benchmarks/tasks/run-benchmark.yml [hunk 2]
```diff
@@ -1,67 +1,90 @@
...
+    body: "{{ lookup('file', archive_dir + '/' + benchmark +'-payload.json') }}"
+    method: POST
+    status_code: 201
+    body_format: json
+
+- name: Cleanup kube-burner namespace
+  shell: |
+    export KUBECONFIG={{kubeconfig}}
+    kubectl delete ns/pod-density-{{benchmark_uuid}}
+  when: item.name == "agent-pod-density"
```
Comment (learnitall) [inline roles/benchmarks/tasks/run-benchmark.yml:90]: Why do we only need to cleanup the kube-burner namespace for using agent-pod-density?

File: roles/benchmarks/tasks/run-benchmark.yml [hunk 3]
```diff
@@ -1,67 +1,90 @@
...
+    benchmark_uuid: "{{ uuid.stdout }}"
+    benchmark: "{{item.name}}"
 
-  - name: Capture results
-    shell: |
-      cp {{ role_path }}/files/uperf.json {{archive_dir}}/
-      touchstone_compare -url {{es_url}} --config {{archive_dir}}/{{item.report}} -u {{benchmark_uuid}} | tee {{archive_dir}}/{{item.name}}-result.out
-    register: report
+- name: Check for run UUID
+  fail:
```
Comment (learnitall) [inline roles/benchmarks/tasks/run-benchmark.yml:49]: Looking over the playbook, my only concern is that we could fail here before the metadata is sent to ES. Do we only want to collect metadata for benchmarks that are successful?

**Review by learnitall — COMMENTED** [review]
Just one concern about shipping the metadata on a failed benchmark, but other than that looks good to go. I'll give it a run and report back

File: roles/benchmarks/tasks/run-benchmark.yml (hunk 2 above)
Comment (jtaleric) [reply roles/benchmarks/tasks/run-benchmark.yml:90]: so, kube-burner will cleanup before the next run... just reduce overall runtime for subsequent runs I just clean up.

File: roles/benchmarks/tasks/run-benchmark.yml (hunk 3 above)
Comment (jtaleric) [reply roles/benchmarks/tasks/run-benchmark.yml:49]: I need to think through this a bit more. Either we can move up the pushing the metadata (before the failure) or simply not fail and just report to ES... I don't have a strong opinion.

File: roles/benchmarks/tasks/run-benchmark.yml (hunk 2 above)
Comment (learnitall) [reply roles/benchmarks/tasks/run-benchmark.yml:90]: Do we need to do this for datapath? Or is it expected that agent runs before datapath?

File: roles/benchmarks/tasks/run-benchmark.yml (hunk 3 above)
Comment (learnitall) [reply roles/benchmarks/tasks/run-benchmark.yml:49]: IMO kind of a mix. Reporting to ES even during a failure would be good for our metrics and I think it's important we still fail the playbook if the benchmark isn't successful as then we can track the failure correctly in the CI.

**Review by learnitall — APPROVED** [review]
LGTM!

**Pull request merged**